core-bundle/src/Security/Authentication/Token/TokenChecker.php

```diff
@@ -97,14 +101,36 @@ public function getBackendUsername(): ?string
         return $token->getUser()->getUserIdentifier();
     }
 
+    /**
+     * Tells whether the front end preview can be accessed.
+     */
+    public function isPreviewAllowed(): bool
```

Comment:
I would prefer `$tokenChecker->hasValidPreviewToken()` instead of `isPreviewAllowed`. WDYT?

Reply:
I’m not sure. A backend user does not have a valid PreviewToken but is still allowed to view the preview. So correct would be `hasValidPreviewTokenOrIsBackendUser()`

Reply:
Maybe naming it similar to `hasBackendUser()` and `hasFrontendUser()`?
Like `hasPreview()`?

Reply:
Or `canAccessPreview`? It should be "speaking" when used together with the `$tokenChecker` variable, so maybe `$tokenChecker->allowsPreview()`?

Reply:
Done in 11a9101049a335767f3775f26544c23a2d5f025a